ntp-proto/src/filter.rs

```diff
@@ -451,6 +444,116 @@ fn find_interval(chime_list: &[CandidateTuple]) -> Option<(NtpDuration, NtpDurat
     None
 }
 
+/// Discard the survivor with maximum selection jitter until a termination condition is met.
+fn cluster_algorithm(candidates: &mut Vec<SurvivorTuple>) {
+    // sort the candidates by increasing lambda_p (the merit factor)
+    candidates.sort_by(|a, b| a.metric.cmp(&b.metric));
+
+    let mut qmax_index = 0;
+
+    let mut min_peer_jitter = 2.0e9;
+    let mut max_selection_jitter = -2.0e9;
+
+    loop {
+        // for each candidate
+        for (index, candidate) in candidates.iter().enumerate() {
+            let p = candidate.peer;
+
+            min_peer_jitter = p.statistics.jitter.min(min_peer_jitter);
```

Comment:
Nit: why not calculate this outside the loop with iterator::min?

Reply:
If you're looping anyway, it's probably better to do everything in one loop rather than looping twice.

Reply:
yeah I don't have strong feelings either way, but that was my thinking.

Reply:
Style suggestion: I often find the asymmetric `.min()` and `.max()` relatively hard to reason about. I've started using `Ord::min()` and `Ord::max()` instead.

Reply:
wait I see now. This minimum is for the whole list of candidates, so it can be computed outside of both loops

Reply:
oh, nice! we need to use `f64::min` here because floats are not `Ord`, but yes I like this